token/cli/src/main.rs

```diff
@@ -1671,15 +1671,16 @@ async fn command_gc(
     for (token_pubkey, accounts) in accounts_by_token.into_iter() {
         println_display(config, format!("Processing token: {}", token_pubkey));
 
-        let token = token_client_from_config(config, &token_pubkey);
+        let token =
+            token_client_from_config(config, &token_pubkey, accounts.values().next().map(|a| a.1));
```

Comment:
nit: Rather than pulling out the decimals here, how about making the key for the outer hashmap `(Pubkey, u8)` for the mint pubkey and the decimals.  Then you'll have the decimals available in line 1671 as:

```
for ((token_pubkey, decimals), accounts) in accounts_by_token.into_iter() {
```

And you won't need to store the decimals in the value for each account in the inner hashmap. What do you think?